avocado/utils/disk.py

```diff
@@ -170,3 +171,94 @@ def is_root_device(device):
         if item['mountpoint'] == "/" and device == str(item['pkname']):
             return True
     return False
+
+
+def is_disk_mounted(device):
+    """
+    check if given disk is mounted or not
+
+    :param device: disk/device name
+    :type device: str
+
+    :returns: True if the device/disk is mounted else False
+    :rtype: bool
+    """
+    with open('/proc/mounts') as mounts:
+        for mount_line in mounts.readlines():
+            dev, fs_dir, fs_vfstype, mount_op, _, _ = mount_line.split()
```

Comment:
does this work if the input disk is partition instead of whole disk and if not all the partitions mounted ?

Reply:
> does this work if the input disk is partition instead of whole disk and if not all the partitions mounted ?

@abdhaleegit yes it works for partition even. Basically if the partition/disk (anything) is formated with some FS and mounted, it works as expected.
logs for your reference

`[root@ltcrain6yo-lp6 ~]# fdisk /dev/mapper/mpathc
[root@ltcrain6yo-lp6 ~]# fdisk -l /dev/mapper/mpathc
Disk /dev/mapper/mpathc: 50 GiB, 53687091200 bytes, 104857600 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 32768 bytes / 32768 bytes
Disklabel type: dos
Disk identifier: 0xe22cb8b7

Device              Boot    Start      End  Sectors  Size Id Type
/dev/mapper/mpathc1           256 20480255 20480000  9.8G 83 Linux
/dev/mapper/mpathc2      20482048 40962047 20480000  9.8G 83 Linux
/dev/mapper/mpathc3      40962048 61442047 20480000  9.8G 83 Linux
/dev/mapper/mpathc4      61442048 81922047 20480000  9.8G 83 Linux
[root@ltcrain6yo-lp6 ~]# 
[root@ltcrain6yo-lp6 ~]# mkfs.xfs /dev/mapper/mpathc1
mkfs.xfs: Volume reports stripe unit of 32768 bytes and stripe width of 0, ignoring.
meta-data=/dev/mapper/mpathc1    isize=512    agcount=4, agsize=640000 blks
         =                       sectsz=512   attr=2, projid32bit=1
         =                       crc=1        finobt=1, sparse=1, rmapbt=0
         =                       reflink=1
data     =                       bsize=4096   blocks=2560000, imaxpct=25
         =                       sunit=0      swidth=0 blks
naming   =version 2              bsize=4096   ascii-ci=0, ftype=1
log      =internal log           bsize=4096   blocks=2560, version=2
         =                       sectsz=512   sunit=0 blks, lazy-count=1
realtime =none                   extsz=4096   blocks=0, rtextents=0
[root@ltcrain6yo-lp6 ~]# 
[root@ltcrain6yo-lp6 ~]# blkid | grep -i mpathc1
/dev/mapper/mpathc1: UUID="a2c1fd4a-7e1f-4d89-976d-e1e701941f9d" BLOCK_SIZE="512" TYPE="xfs" PARTUUID="e22cb8b7-01"
[root@ltcrain6yo-lp6 ~]# 
[root@ltcrain6yo-lp6 ~]# mkdir mount_point
[root@ltcrain6yo-lp6 ~]# mount /dev/mapper/mpathc1 /root/mount_point
[root@ltcrain6yo-lp6 ~]# mount | grep -i point
/dev/mapper/mpathc1 on /root/mount_point type xfs (rw,relatime,seclabel,attr2,inode64,logbufs=8,logbsize=32k,noquota)
[root@ltcrain6yo-lp6 ~]# 
>>> disk.is_dir_mounted('/root/mount_point')
True
>>>
>>> disk.is_dir_mounted('/root/mount_')
False
>>>
>>> disk.is_disk_mounted('/dev/mapper/mpathc1')
True
>>> 
>>> disk.is_disk_mounted('/dev/mapper/mpathc2')
False
>>> 
>>> disk.is_fs_exists('/dev/mapper/mpathc1')
True
>>> disk.is_fs_exists('/dev/mapper/mpathc2')
False
>>> 
>>> disk.get_disk_mountpoint('/dev/mapper/mpathc1')
'/root/mount_point'
>>> 
>>> disk.get_dir_mountpoint('/root/mount_point')
'/dev/mapper/mpathc1'
>>> 
`

Reply:
@Naresh-ibm ok,thanks